pkg/istio-agent/xds_proxy.go

```diff
@@ -105,13 +108,15 @@ func initXdsProxy(ia *Agent) (*XdsProxy, error) {
 		LocalHostAddr: localHostAddr,
 	}
 	proxy := &XdsProxy{
-		istiodAddress:  ia.proxyConfig.DiscoveryAddress,
-		clusterID:      ia.secOpts.ClusterID,
-		localDNSServer: ia.localDNSServer,
-		stopChan:       make(chan struct{}),
-		healthChecker:  health.NewWorkloadHealthChecker(ia.proxyConfig.ReadinessProbe, envoyProbe),
-		xdsHeaders:     ia.cfg.XDSHeaders,
-		xdsUdsPath:     ia.cfg.XdsUdsPath,
+		istiodAddress:               ia.proxyConfig.DiscoveryAddress,
+		clusterID:                   ia.secOpts.ClusterID,
+		localDNSServer:              ia.localDNSServer,
+		stopChan:                    make(chan struct{}),
+		healthChecker:               health.NewWorkloadHealthChecker(ia.proxyConfig.ReadinessProbe, envoyProbe),
+		xdsHeaders:                  ia.cfg.XDSHeaders,
+		xdsUdsPath:                  ia.cfg.XdsUdsPath,
+		xdsEnableTokenFetchExchange: ia.secOpts.XdsEnableTokenFetchExchange,
```

Comment:
How about: 
- add a field 'secOpts' to XdsProxy
- use it ( without having to duplicate all fields inside )
We may use other settings.

Reply:
XdsProxy has six other fields coming from secOpts. To keep this PR focused and smaller, the refactoring of XdsProxy fields can be done in a separate PR.